### PR TITLE
Gorealis Resiliency

### PR DESCRIPTION
**Pull request by kkrishna**

### CODE REVIEW

File: realis.go [hunk 1]
```diff
@@ -48,25 +50,143 @@ type Realis interface {
...
+	config.password = passwd
+	config.cluster = cluster
+	config.url = ""
+	// Configured for vagrant
+	AddBasicAuth(config, user, passwd)
+	r := newClient(config)
+	return r, nil
+}
+
+func GetDefaultClusterFromZKUrl(zkurl string) *Cluster {
```
Comment (rdelval) [inline realis.go]: Nit: Just by Go convention, probably a good idea to get rid of "Get", but overall it's not a big deal.

File: realis.go [hunk 2]
```diff
@@ -92,41 +212,43 @@ func defaultTTransport(urlstr string, timeoutms int) (thrift.TTransport, error)
...
 	httpTrans.SetHeader("Content-Type", "application/vnd.apache.thrift.binary")
 	httpTrans.SetHeader("User-Agent", "GoRealis v1.0.4")
 
-	return RealisConfig{transport: trans, protoFactory: thrift.NewTBinaryProtocolFactoryDefault()}, nil
+	return &RealisConfig{transport: trans, protoFactory: thrift.NewTBinaryProtocolFactoryDefault()}, nil
 
 }
 
 // Helper function to add basic authorization needed to communicate with Apache Aurora.
 func AddBasicAuth(config *RealisConfig, username string, password string) {
```
Comment (rdelval) [inline realis.go]: Any chance we can get away with just using a realisConfig now that username and password are internal?

File: realis.go [hunk 3]
```diff
@@ -165,90 +353,217 @@ func (r *realisClient) GetInstanceIds(key *aurora.JobKey, states map[aurora.Sche
...
+			time.Sleep(adjusted)
+			duration = time.Duration(float64(duration) * defaultBackoff.Factor)
+		}
 
-	if err != nil {
-		return nil, errors.Wrap(err, "Error sending Create command to Aurora Scheduler")
+		if resp, err = r.client.CreateJob(auroraJob.JobConfig()); err == nil {
+			return response.ResponseCodeCheck(resp)
+		}
+		err1 := r.ReestablishConn()
```
Comment (rdelval) [inline realis.go]: Should there be an else here?

Comment (rdelval) [inline realis.go]: Whoops scratch that, this falls through only if it fails

Comment (kkrishna) [inline realis.go]: yep keeping it simple.. improves readablilty, keeping logic with minimal indentation.

File: realis.go (hunk 2 above)
Comment (kkrishna) [inline realis.go]: yep this can be made private.. will be taken care in next pr.

File: realis.go (hunk 1 above)
Comment (kkrishna) [inline realis.go]: ok

**Pull request merged**